Add people to export result

rmathew1011 opened the pull request:
# Description

Resolves #505 
 - Adds `People` to export results.  
 - Raw data is now displayed as intended - example, the literal: `FacultyMember` appears as `Faculty Member`, `NonFacultyAcademic` as `Non Faculty Academic`, and `GraduateStudent` as `Student Researcher`.

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)
# Testing Procedures

Tested on local set up.
below are the 3 scenarios:
Faculty Member
<img width="1522" height="333" alt="Screenshot 2025-11-10 at 12 07 56 PM" src="https://github.com/user-attachments/assets/3dccfd9a-2d7e-4974-a269-2e537d685cd9" />

Non Faculty Academic
<img width="1221" height="144" alt="Screenshot 2025-11-10 at 12 03 57 PM" src="https://github.com/user-attachments/assets/095a21cd-d80f-4b26-92df-eb83d5602788" />

Student Researcher
<img width="1151" height="131" alt="Screenshot 2025-11-10 at 12 00 22 PM" src="https://github.com/user-attachments/assets/c95f9d33-3bf9-4873-a8aa-e3185c3b6c78" />

rmathew1011 committed:
Resolves #505
 - Adds People type formatting to export results.
 - Raw data is now displayed as intended - example, the literal: `FacultyMember` appears as `Faculty Member`, `NonFacultyAcademic` as `Non Faculty Academic`, and `GraduateStudent` as `Student Researcher`.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/config/model/ExportConfig.java:
```diff
@@ -16,6 +19,8 @@ public class ExportConfig {
 
     private String individualBaseUri = "http://localhost:4200/display";
 
+    private Map<String, String> personTypeMapping = new HashMap<>();
```
```suggestion
    private Map<String, String> personTypes = new HashMap<>();
```

I suggest renaming this to a shorter name, like `personTypes`.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/config/model/ExportConfig.java:
```diff
@@ -32,4 +37,18 @@ public void setIndividualBaseUri(String individualBaseUri) {
+            personTypeMapping.put("NonFacultyAcademic", "Non Faculty Academic");
+        }
+        return personTypeMapping;
+    }
```
This is a configuration file (See the [config props annotation](https://github.com/TAMULib/scholars-discovery/blob/76f9384305133be3d7bdfd5e407bb51f131d7a15/src/main/java/edu/tamu/scholars/middleware/config/model/ExportConfig.java#L15)).

The mapping should not be initialized in a getter.

Instead, the configuration properties handles this.
These settings would then be defined in the [application.yml](https://github.com/TAMULib/scholars-discovery/blob/76f9384305133be3d7bdfd5e407bb51f131d7a15/src/main/resources/application.yml#L4).

The getter would just return the value as-is.

Then update the documentation/readme appropriately to describe these new settings.

rmathew1011 replied:
Added the configuration data to the export section in application yml file. Cleaned up the getter method.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/export/service/CsvExporter.java:
```diff
@@ -121,6 +125,17 @@ private List<Object> getRow(
+                if (property.equals("type")) {
+
+                    @SuppressWarnings("unchecked")
+                    List<String> values = (List<String>) value;
```
This is repeated in multiple places in the file and the Suppress Warnings annotation is an exceptional situation.

I suggest to make a private method that does this (along with the suppress warnings annotation) and then call that function to get the value as a list.

### Quick Example
```java
    @SuppressWarnings("unchecked")
    private List<String> asStringList(Object value) {
       return (List<String>) value;
    }
```

kaladay commented on src/main/java/edu/tamu/scholars/middleware/export/service/CsvExporter.java:
```diff
@@ -150,4 +165,25 @@ private String serialize(String value) {
             : value;
     }
 
+    private String capitalize(String text) {
```
I suggest removing this entirely and relace it with `StringUtils.capitalize()`.

The [Apache Lang3 dependency is already included](https://github.com/TAMULib/scholars-discovery/blob/76f9384305133be3d7bdfd5e407bb51f131d7a15/pom.xml#L330) and this dependency provides `StringUtils`.

see: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html
> Capitalizes a String changing the first character to title case as per [Character.toTitleCase(int)](https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#toTitleCase-int-).

This might also have to be called **after** the word-splitting.

There is also a potentially better option than `StringUtils.capitalize()`.
And that is `WordUtils.capitalizeFully()`.

see: https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/WordUtils.html
> Converts all the whitespace separated words in a String into capitalized words, that is each word is made up of a titlecase character and then a series of lowercase characters.

rmathew1011 replied:
Utilized WordUtils.capitalizeFully() - this cleaned up the code a lot. Thank you so much.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/export/service/CsvExporter.java:
```diff
@@ -150,4 +165,25 @@ private String serialize(String value) {
+            return type;
+        }
+        String mapped = config.getPersonTypeMapping().get(type);
+        if (!mapped.equals(type)) return mapped;
```
```suggestion
        if (config.getPersonTypeMapping().containsKey(type)) return config.getPersonTypeMapping().get(type);
```

The `type` is the key and the return of `.get(type)` is the value at the key.
This should not be comparing the map key to the map value.

Instead, check if the key exists and if it does, then return the value as-is.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/export/service/CsvExporter.java:
```diff
@@ -150,4 +165,25 @@ private String serialize(String value) {
+                        .map(this::capitalize)
+                        .collect(Collectors.joining(SPACE_STRING));
+
+        return result;
```
```suggestion
        return Arrays.stream(type.replaceAll("([a-z])([A-Z])", "$1 $2").split(SPACE_STRING))
            .map(this::capitalize)
            .collect(Collectors.joining(SPACE_STRING));
```

This is the last call in the function.
It is cleanier to just return the result directly.

Also, reduce the excessive spacing.
It should be 4 space tabbing, based on the rest of the file.

I will also note that the `[a-z])([A-Z]` regex is not multi-lingual safe.
It only handles US ASCII characters.

We should do some research to find a multi-lingual friendly way to do word splitting.

rmathew1011 replied:
Will create note to address this soon.

rmathew1011 committed:
Update src/main/java/edu/tamu/scholars/middleware/config/model/ExportConfig.java

Co-authored-by: Kevin Day <35114839+kaladay@users.noreply.github.com>

rmathew1011 committed:
Code cleanup based on PR review change request
 - added properties to the export settings in the configuration file.
 - encapsulated the unchecked cast in a seperate method to avoid repetition.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/export/service/CsvExporter.java:
```diff
@@ -121,11 +121,20 @@ private List<Object> getRow(
                 Object value = content.get(property);
+                if (property.equals("type")) {
+
+                    List<String> values = asStringList(value);
```
```suggestion
                    List<String> values = asStringList(value);
```

I suggest this style clean up by removing the extra line.

kaladay commented on src/main/java/edu/tamu/scholars/middleware/export/service/CsvExporter.java:
```diff
@@ -121,11 +121,20 @@ private List<Object> getRow(
 
-                    @SuppressWarnings("unchecked")
-                    List<String> values = (List<String>) value;
+                    List<String> values = asStringList(value);
```
```suggestion
                    List<String> values = asStringList(value);
```
I suggest this style clean up by removing the extra line.

rmathew1011 replied:
Unable to apply this suggestion.

kaladay reviewed (approved):
Approved with some style suggestions.

rmathew1011 committed:
Removed extra lines

rmathew1011 committed:
Removed extra lines
